frontend/src/pages/HomePage.tsx

```diff
@@ -26,6 +27,7 @@ export const HomePage = () => {
   useEffect(() => {
     async function signInSignUp() {
       const response = await requestMSAuthResult(instance, accounts[0]);
+      persistToken(response.idToken)
```

Comment:
This shouldn't be in a tsx file - we should try as much as possible to keep tsx files for visible stuff like buttons, textfields etc. Let's move this into the end of the requestMSAuthResult function. That way, every time we call requestMSAuthResult we can be sure it updates localStorage (in the future, we may re-use this function).